app/controllers/concerns/application_shared.rb

```diff
@@ -15,4 +15,35 @@ def set_timezone
     # Reset timezone for the next thread
     Time.zone = default_timezone
   end
-end
\ No newline at end of file
+
+  private
+
+  # Get the allowed parameters from the Apipie documentation in the API controllers.
+  def safe_params
+    apipie_params = Apipie.app.get_method_description(controller_name, action_name).to_json[:params]
```

Comment:
1) I'm fine with this approach, but it's an 80/20 kind of situation where the API controller will help with 80% of the attributes, and the remainder we will have to specify by wrapping this method on a per-controller basis.
2) In dev I'm struggling to get this to work, as get_method_description keeps returning nil even after using the controller, but it's fine in test.  Perhaps something to do with autoloading?  (Though still doesn't work even after using that API.)